articles/session-management-in-nodejs-using-expressjs-and-express-session/index.md

```diff
@@ -0,0 +1,270 @@
+const mypassword = 'mypassword'
+
+const PORT = 4000;
+var session;
+const app = express();
+
+// get our app to use body parser 
+app.use(bodyParser.urlencoded({ extended: true }))
+
+// cookie parser middleware
+app.use(cookieParser());
+
+// serve css styling
+app.use(express.static(__dirname));
+
+// creating 24 hours from milliseconds
+const oneDay = 1000 * 60 * 60 * 24;
+
+//session middleware
+app.use(sessions({
+    secret: "thisismysecrctekeyfhrgfgrfrty84fwir767",
+    saveUninitialized:true,
+    cookie: { maxAge: oneDay },
+    resave: false 
+}));
+
+app.get('/',(req,res) => {
+    session=req.session;
+    if(session.userid){
+        res.send("welcome User <a href=\'/logout'>click to logout</a>");
+    }else
+    res.sendFile('index.html',{root:__dirname})
+});
+
+app.post('/user',(req,res) => {
+    if(req.body.username == myusename && req.body.password == mypassword){
+        session=req.session;
+        session.userid=req.body.username;
+        console.log(req.session)
+        res.send(`Hey there, welcome <a href=\'/logout'>click to logout</a>`);
+    }
+    else{
+        res.send('invalid username or password');
+    }
+})
+
+app.get('/logout',(req,res) => {
+    req.session.destroy();
+    res.redirect('/');
+});
+
+app.listen(PORT, () => console.log(`Server Running at port ${PORT}`));
+```
+
+In this example, we are using a simple login application. To authenticate the user, I've specified the username and password as `user1` and `mypassword`, respectively. In a production environment, these credentials are usually saved in a database. For the sake of simplicity in this tutoirial, we are storing them in these variables. The server will verify the username and password against these values. If valid, the user will be granted the necessary access.
+
+To set this session, the user will submit the credentials. The server will verify these credentials as set `req.body.username == myusername && req.body.password == mypassword`. I.e., comparing the username and the password for the existing user.
+
+If the credentials are valid, the server will create a temporary user session with a random string known as a session ID to identify that session.
+
+The server will send a cookie to the client's browser. The session ID is going to be placed inside this cookie.
+
+Once the client browser saves this cookie, it's then going to send that cookie along with each subsequent request to the server.
+
+The server is going to validate the cookie against the session ID. If the validation is successful, the user is granted access to the requested resources on the server.
+
+If the credentials are invalid, the server will not grant this user access to the resources. No session will be initialized, and no cookie will be saved.
+
+When the user decides to log out, the server will destroy (`req.session.destroy();`) the session and clear out the cookie on the client-side. Cookies are cleared in the browser when the `maxAge` expires.
```

Comment:
Hi @Catemacharia. This part of the tutorial is not well structured. 

You are giving the entire code at once and then talking about what the code does. But the reader will not know which part of the code is doing which step. 

Please split the code and explain how to build the JS file step-by-step instead of giving it all at once. Please take a look at other articles to better understand how to structure this. 

For example, take a look at [this article](https://www.section.io/engineering-education/text-to-speech-in-javascript/) that has been already published.